File: app/src/main/java/engine/graphics/DrawRectCall.java
package engine.graphics;

import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.RectF;

public class DrawRectCall implements DrawCall {
    RectF dst;
    Paint paint;

    public DrawRectCall(RectF dst, Paint paint) {
        this.dst = dst;
        this.paint = paint;
    }

    @Override
    public void draw(Canvas canvas) {
        canvas.drawRect(dst, paint);
    }
}

File: engine2d/src/main/java/life/unmedicated/engine2d/graphics/DrawRectCall.kt
package life.unmedicated.engine2d.graphics

import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.RectF

class DrawRectCall(private val rect: RectF, private val paint: Paint) : DrawCall {
    override fun draw(canvas: Canvas) {
        canvas.drawRect(rect, paint)
    }
}